fili-core/src/test/groovy/com/yahoo/bard/webservice/data/cache/MemDataCacheSpec.groovy

```diff
@@ -45,4 +49,42 @@ class MemDataCacheSpec extends Specification {
         0 * future.get()
         result
     }
+
+    def "when #waitForFuture is set to false , It doesn't wait for future anymore"() {
+
+        SYSTEM_CONFIG.setProperty(MemDataCache.WAIT_FOR_FUTURE, waitForFuture as String)
```

Comment:
You can manipulate the property through the FeatureFlag object.
Also, don't forget to reset in a cleanup block.